app/models/user_optional_field.rb

```diff
@@ -0,0 +1,13 @@
+class UserOptionalField < ApplicationRecord
+  belongs_to :user
+
+  validates :label, presence: true, length: { maximum: 30 }, uniqueness: { scope: :user_id }
+  validates :value, presence: true, length: { maximum: 128 }
+  validate :validate_quota
+
+  def validate_quota
+    return unless user
+
+    errors.add(:user_id, "already has the maximum allowable optional fields") unless user.user_optional_fields.count < 3
```

Comment:
I see that @benhalpern mentioned caping the number of optional fields at 3 in his comment at https://github.com/thepracticaldev/dev.to/issues/2365#issuecomment-496982227, but that was referring to an implementation where each key and value was a named column on `users`. Since these values have been normalized to `user_optional_fields`, is this limit still desired?